evalml/pipelines/components/transformers/samplers/oversampler.py

```diff
@@ -67,17 +67,34 @@ def fit(self, X, y):
             self
         """
         X_ww, y_ww = self._prepare_data(X, y)
+
         sampler_name = self._get_best_oversampler(X_ww)
         self.sampler = self.sampler_options[sampler_name]
 
         # get categorical features first, if necessary
         if sampler_name == "SMOTENC":
-            self._get_categorical(X)
-        super().fit(X, y)
+            self._get_categorical(X_ww)
+        super().fit(X_ww, y_ww)
         return self
 
+    def transform(self, X, y=None):
```

Comment:
this is where my 10000 open PRs is biting me in the butt! tl;dr - we'll eventually need the handle method back in the Oversampler's transform, so I didn't feel like deleting this since it's not going into main.

Full explanation:

You're correct that this is doing nothing. Previously, I needed the `_handle_nullable_types` call because of an incompatibility in `y`, but that was resolved by sklearn 1.2.2, so I removed it and must've forgotten to remove this method. 

But we also have a new oversampler incompatibility in `X` ([this PR](https://github.com/alteryx/evalml/pull/4068)) that will eventually mean we need a call to _handle_nullable_types in this method, which'll mean we need this method. But since _this_ is the PR to add the calls to the handle method, we can't yet add it in the other PR.